### PR TITLE
New docs website / Icons page

### PR DESCRIPTION
**Pull request by didoo**
### :pushpin: Summary

This PR implements the `Icons` page of the `Foundation` section, per designs.

### :hammer_and_wrench: Detailed description

In this PR I have:
- added a markdown "doc" page for `Icons` 
- added a `Doc::IconList` component as content to the "Library" section
  - the component has been ported from the FlightIcons website and then refactored and re-skinned to follow the latest design specs
- ported the existing documentation ([Getting started for designers](https://flight-hashicorp.vercel.app/design), [Getting started for engineers](https://flight-hashicorp.vercel.app/engineering)) and associated images, from the FlightIcons website to the "Code", "Guidelines" and "Contributing" sections

👉 👉 👉 **Preview**: Preview: https://hds-website-git-new-docs-website-icons-page-hashicorp.vercel.app/foundations/icons/

### 🚧 TODOs
- [ ] ~~understand why the query parameters are not added to the URL~~
    - opened a specific ticket ([HDS-1216](https://hashicorp.atlassian.net/browse/HDS-1216)) to tackle this issue to avoid this being blocked

### :camera_flash: Screenshots

**In code**
screenshot_2156

**In design** ([Figma file](https://www.figma.com/file/Ky0qWjvHZR3je1lCBlzNB1/HDS-website?node-id=924%3A50478))
screenshot_2155

### :link: External links

JIRA ticket: https://hashicorp.atlassian.net/browse/HDS-931
Figma file: https://www.figma.com/file/Ky0qWjvHZR3je1lCBlzNB1/HDS-website?node-id=924%3A50478

***

### 👀 How to review

👉 Review commit-by-commit or by files changed

:speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.

### CODE REVIEW

File: website/app/components/doc/icons-list/meta-row.hbs
Comment (Dhaulagiri) [inline website/app/components/doc/icons-list/meta-row.hbs:4]: May be unavoidable, but allowing people to copy the keywords doesn't seem that useful
Screen Shot 2022-12-09 at 11 43 00

Given how much space things are taking up in the design I wonder if they are actually worth showing at all? They're more of an internal thing than something consumers would care about.

Comment (didoo) [reply website/app/components/doc/icons-list/meta-row.hbs]: Pushed

**Review by Dhaulagiri — APPROVED** [review]
I think the code here looks good. It'll be easier to work on the overall query params question in the other PR once this lands.

**Comment by Dhaulagiri** [comment]
@didoo can this be merged? I need to build a couple of follow ups on top of this.

**Comment by didoo** [comment]
> @didoo can this be merged? I need to build a couple of follow ups on top of this.

@Dhaulagiri I was waiting for @heatherlarsen @cveigt approval. Merging now.

**Comment by cveigt** [comment]
@didoo I didn't get to review it before it was merged; sorry. Is there another PR or place where I could leave comments?

**Comment by didoo** [comment]
> @didoo I didn't get to review it before it was merged; sorry. Is there another PR or place where I could leave comments?

Comment in this PR (in the code tab, if you want) and I'll make changes in a follow-up PR

**Pull request merged**
